cirq-core/cirq/qis/clifford_tableau.py

```diff
@@ -22,7 +23,37 @@
     import cirq
 
 
-class CliffordTableau:
+class StabilizerState(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def apply_x(self, g: 'cirq.XPowGate', axis: int):
```

Comment:
Probably pass exponent and phase here instead of the gate. That way we don't have to create a dependency (even indirectly) on `cirq.ops` here.